src/codegen/statements.rs

```diff
@@ -1495,7 +1496,7 @@ impl Type {
             Type::Array(ty, dims) => {
                 ty.default(ns)?;
 
-                if dims[0].is_none() {
+                if dims[0] == ArrayLength::Dynamic {
```

Comment:
There are a bunch of these which are wrong.

I thought about fixing them in another PR but why not here. I've fixed all of them. This might fix #903